src/plugins/exportAsJSONAction/ExportAsJSONAction.js

```diff
@@ -109,49 +111,41 @@ export default class ExportAsJSONAction {
      * @param {object} parent
      * @returns {object}
      */
-    _rewriteLink(child, parent) {
-        this.externalIdentifiers.push(this._getId(child));
-        const index = parent.composition.findIndex(id => {
-            return _.isEqual(child.identifier, id);
-        });
-        const copyOfChild = JSON.parse(JSON.stringify(child));
+    _rewriteLink(child, parent, isReference) {
+        const originalKeyString = this._getId(child);
+        const parentKeyString = this._getId(parent);
+        const existingMappedKeyString = this.idMap[originalKeyString];
+        let copy;
 
-        copyOfChild.identifier.key = uuid();
-        const newIdString = this._getId(copyOfChild);
-        const parentId = this._getId(parent);
+        if (!existingMappedKeyString) {
+            this.externalIdentifiers.push(originalKeyString);
 
-        this.idMap[this._getId(child)] = newIdString;
-        copyOfChild.location = parentId;
-        parent.composition[index] = copyOfChild.identifier;
-        this.tree[newIdString] = copyOfChild;
-        this.tree[parentId].composition[index] = copyOfChild.identifier;
+            copy = JSON.parse(JSON.stringify(child));
+            copy.identifier.key = uuid();
+            copy.location = isReference ? null : parentKeyString;
```

Comment:
What's the reason for nulling the location here? I understand that we don't want to make the condition set a child of the display layout (for example), but I'm concerned about the implications of explicitly setting it to null. I think it might convert condition sets in the tree to links?

Reply:
Great question! I was wondering the same thing. I think the idea was to assume that the item was outside of the tree? I have no clue to be honest. Setting it to `null` would just "delete" it, but it could still be referenced, it just wouldn't show anywhere in the tree UNLESS it was in the tree already, let's chat on the best way to handle this.

Reply:
In the comments, there was some reference to the fact that the conditional styles aren't included in the composition of the object, so that was the reason they had to be handled separately.

Reply:
@shefalijoshi Any idea why this is nulled?

Reply:
I think it's nulled because a reference (like a condition set referenced in an object style) doesn't have to be in the same location as the parent. It can be anywhere in the tree.
Perhaps instead of null it could be the `root`.

Reply:
Yeah, I see the logic behind that, but there's a chance it could be in the exported tree as well right? just in another location? Looks like we might need to handle for both scenarios, but the interesting one is what to do if it's NOT in the exported tree. I could see just plunking it into the root of the export, but we may need to have a notification or something that let's the user know that happened.

Reply:
Yes, it could be in the exported tree.

Reply:
Yeah, conditional styling is a bad edge case that we have ignored for a while. It creates orphaned condition sets that are unreachable from the tree. 

For now I am fine to say that condition sets referenced in conditional styling HAVE to reside in the exported tree. We already have an issue to address the orphaned condition sets - https://github.com/nasa/openmct/issues/5642

If they are not located in the exported tree you _can_ duplicate the condition set into a new location in the imported environment and update the display to reference that instead. It's a pain, but it's a workaround

Reply:
So I'm not sure we should be nulling that location, because the condition set might be in the exported tree somewhere, and we would be turning it into a link I think?

What happens if we _don't_ null it, and the condition set _does not_ reside within the exported tree? What are we guarding against here?

Reply:
Not null-ing it should be fine. Let's go with that recommendation.